apps/examples/testcase/ta_tc/mqtt/utc/utc_mqtt_main.c

```diff
@@ -0,0 +1,479 @@
+	if (res < 0) {
+		UTC_MQTT_LOGE;
+		return ;
+	}
+
+	UTC_MQTT_WAIT_SIGNAL;
+	
+	res = mqtt_publish(g_mqtt_pub_handle, UTC_MQTT_TOPIC, g_mqtt_msg, sizeof(g_mqtt_msg), 0, 0);
+	TC_ASSERT_EQ("mqtt_publish", res, 0);
+	UTC_MQTT_WAIT_SIGNAL;
+	TC_SUCCESS_RESULT();
+}
+
+
+/**
+ * @testcase         utc_mqtt_unsubscribe_n
+ * @brief            stop subscribing from a broker
+ * @scenario         call mqtt_unsubscribe with an invalid input
+ * @apicovered       mqtt_unsubscribe
+ * @precondition     none
+ * @postcondition    none
+ */
+void utc_mqtt_unsubscribe_n(void)
+{
+	int res = mqtt_unsubscribe(NULL, NULL);
+	TC_ASSERT_LT("mqtt_unsubscribe", res, 0);
+	TC_SUCCESS_RESULT();
+}
+
+
+/**
+ * @testcase         utc_mqtt_unsubscribe_p
+ * @brief            stop subscribing from a broker
+ * @scenario         call mqtt_unsubscribe
+ * @apicovered       mqtt_unsubscribe
+ * @precondition     none
+ * @postcondition    none
+ */
+void utc_mqtt_unsubscribe_p(void)
+{
+	int res = mqtt_unsubscribe(g_mqtt_sub_handle, UTC_MQTT_TOPIC);
+	TC_ASSERT_EQ("mqtt_unsbscribe", res, 0);
+	TC_SUCCESS_RESULT();
+}
+
+
+/**
+ * @testcase         utc_mqtt_disconnect_n
+ * @brief            disconnect to broker
+ * @scenario         disconnect to broker using an invalid handle
+ * @apicovered       mqtt_disconnect
+ * @precondition     none
+ * @postcondition    none
+ */
+void utc_mqtt_disconnect_n(void)
+{
+	int res = mqtt_disconnect(NULL);
+	TC_ASSERT_LT("mqtt_disconnect", res, 0);
+	TC_SUCCESS_RESULT();
+}
+
+
+/**
+ * @testcase         utc_mqtt_disconnect_p
+ * @brief            disconnect to broker
+ * @scenario         disconnect to broker using a valid handle
+ * @apicovered       db_init
+ * @precondition     none
+ * @postcondition    none
+ */
+void utc_mqtt_disconnect_p(void)
+{
+	int res = mqtt_disconnect(g_mqtt_sub_handle);
+	TC_ASSERT_EQ("mqtt_disconnect", res, 0);
+
+	UTC_MQTT_WAIT_SIGNAL;
+	
+	TC_SUCCESS_RESULT();
+}
+
+
+/**
+ * @testcase         utc_mqtt_deinit_client_n
+ * @brief            free all resources used to mqtt stack
+ * @scenario         call mqtt_deinit with an invalid input
+ * @apicovered       mqtt_deinit_client
+ * @precondition     none
+ * @postcondition    none
+ */
+void utc_mqtt_deinit_client_n(void)
+{
+	int res = mqtt_deinit_client(NULL);
+	TC_ASSERT_LT("mqtt_deinit_client", res, 0);
+	TC_SUCCESS_RESULT();
+}
+
+
+/**
+ * @testcase         utc_mqtt_deinit_client_p
+ * @brief            free all resources used to mqtt stack
+ * @scenario         call mqtt_deinit with an invalid input
+ * @apicovered       mqtt_deinit_client
+ * @precondition     none
+ * @postcondition    none
+ */
+void utc_mqtt_deinit_client_p(void)
+{
+	int res = mqtt_deinit_client(g_mqtt_pub_handle);
+	if (res < 0) {
+		UTC_MQTT_LOGE;
+	}
+	
+	res = mqtt_deinit_client(g_mqtt_sub_handle);
+	TC_ASSERT_EQ("mqtt_deinit_client", res, 0);
+	TC_SUCCESS_RESULT();
+}
+
+
+int mqtt_utc(int argc, FAR char *argv[])
+{
+	int res = _utc_mqtt_init();
```

Comment:
1. You didn't initialize the *total_pass* and *total_fail*. That is needed at start of tc.
2. If you didn't use semaphore, before finishing this, new tc will be launched. Let's use it like belows.
3. We should print the result of tc.
```
int kernel_tc_main(int argc, char *argv[])
{
	sem_wait(&tc_sem);
	working_tc++;

	total_pass = 0;
	total_fail = 0;

	printf("=== TINYARA Kernel TC START! ===\n");
...
...
	printf("\n=== TINYARA Kernel TC COMPLETE ===\n");
	printf("\t\tTotal pass : %d\n\t\tTotal fail : %d\n", total_pass, total_fail);

	working_tc--;
	sem_post(&tc_sem);

	return 0;
}
```

Reply:
done, Thanks